crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -172,30 +172,90 @@ impl Inner {
             block_number: event.block_number,
             log_index: event.log_index,
             auction_id,
-            auction_data,
+            auction_data: auction_data.clone(),
         };
 
         tracing::debug!(?hash, ?update, "updating settlement details for tx");
 
         {
             // temporary to debug and compare with current implementation
-            // TODO: use instead of current implementation
-            let Ok(transaction) = self.eth.transaction(hash.into()).await else {
-                tracing::warn!(?hash, "transaction not found");
-                return Ok(true);
-            };
-            let domain_separator = self.eth.contracts().settlement_domain_separator();
             let settlement = domain::settlement::Settlement::new(
                 transaction,
                 domain_separator,
                 &self.persistence,
             )
             .await;
 
-            tracing::info!(
-                "settlement object {:?}",
-                settlement.map(|settlement| (settlement.observation()))
-            );
+            // automatic checks vs current implementation
+            match (settlement, auction_data) {
+                (Ok(_), None) => {
+                    // bug: we should have an auction_data
+                    tracing::warn!(?auction_id, "automatic check error: missing auction_data");
+                }
+                (Ok(settlement), Some(auction_data)) => {
+                    // staging settlement properly built
+                    let observation = settlement.observation();
+                    if observation.surplus.0 != auction_data.surplus {
+                        tracing::warn!(
+                            ?auction_id,
+                            ?observation.surplus,
+                            ?auction_data.surplus,
+                            "automatic check error: surplus mismatch"
+                        );
+                    }
+                    if observation.fee.0 != auction_data.fee {
+                        tracing::warn!(
+                            ?auction_id,
+                            ?observation.fee,
```

Comment:
nit: I belive we don't use dot-separated log fields anywhere. Also, in different parts of the code the same variable is called differently. For example, this specific is logged both as `observation.fee` and `observation_fee`. Another example: `auction_data.fee` and `fee`. Maybe it makes sense to use underscore everywhere.

```suggestion
                            observation_fee = ?observation.fee,
```